publish-mqtt/src/main.rs

```diff
@@ -420,42 +422,31 @@ async fn connect_sensor_at_idx(
     Ok(())
 }
 
-async fn connect_start_sensor<'a>(
+async fn connect_and_subscribe_sensor_or_disconnect<'a>(
     session: &MijiaSession,
-    sensor: &mut Sensor,
+    id: &DeviceId,
 ) -> Result<(), anyhow::Error> {
-    println!("Connecting from status: {:?}", sensor.connection_status);
     session
         .bt_session
-        .connect(&sensor.id)
+        .connect(id)
         .await
         .with_context(|| std::line!().to_string())?;
-    match session.start_notify_sensor(&sensor.id).await {
-        Ok(()) => Ok(()),
-        Err(e) => {
-            // If starting notifications failed a second time, disconnect so
-            // that we start again from a clean state next time.
-            match sensor.connection_status {
-                ConnectionStatus::Unknown
-                | ConnectionStatus::Connecting { .. }
-                | ConnectionStatus::Disconnected
-                | ConnectionStatus::MarkedDisconnected
-                | ConnectionStatus::WatchdogTimeOut => {
-                    sensor.connection_status = ConnectionStatus::SubscribingFailedOnce;
-                }
-                ConnectionStatus::SubscribingFailedOnce => {
-                    session
-                        .bt_session
-                        .disconnect(&sensor.id)
-                        .await
-                        .with_context(|| std::line!().to_string())?;
-                    sensor.connection_status = ConnectionStatus::Disconnected;
-                }
-                ConnectionStatus::Connected => panic!("This should never happen."),
-            };
+
+    let mut backoff = ExponentialBackoff::default();
+    backoff.max_elapsed_time =
+        Some(SENSOR_CONNECT_RESERVATION_TIMEOUT - 2 * DBUS_METHOD_CALL_TIMEOUT);
```

Comment:
I'd rather this just be a constant which is enough small enough than `SENSOR_CONNECT_RESERVATION_TIMEOUT`, rather than making DBUS_METHOD_CALL_TIMEOUT public.